### PR TITLE
I385

### PR DESCRIPTION
**Pull request by richfitz**
This fixes 

* https://vimc.myjetbrains.com/youtrack/issue/VIMC-385
* https://vimc.myjetbrains.com/youtrack/issue/VIMC-250

and the related PR in montagu-ci fixes https://vimc.myjetbrains.com/youtrack/issue/VIMC-966

This hugely simplifies the deployment of the registry, though I think we've forgotten the faff with the self signed certificates.

docker auth is _super_ basic and it's an all-or-nothing deal.  So all operations (both pull and push) require authentication.  docker keeps the auth floating around for ages so this should not practically be that annoying (I did look into separate read/write permissions and it requires setting up another nginx proxy in front of the registry which seems like more faff than we want right now).

I have tested this on support by using a separate container (port 5001, name registry2, data volume registry_data2).  This is easy to poke about with if you fix the run.sh file.

### CODE REVIEW

**Comment by richfitz** [comment]
(if you could both look over this that'd be terrific because this has security implications)

File: scripts/docker_clone_volume.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
```
Comment (hillalex) [inline scripts/docker_clone_volume.sh:1]: is this file needed/used?

Comment (richfitz) [reply scripts/docker_clone_volume.sh:1]: not needed now but when we were getting going I used it a couple of times to duplicate data volumes.  But yeah, it's fairly googleable

Comment (hillalex) [reply scripts/docker_clone_volume.sh:1]: cool, no harm in keeping it around, maybe just a note in the README to explain what it is!

**Comment by hillalex** [comment]
This all looks good to me, and I agree, having separate read/write permissions isn't important at this stage!

**Review by hillalex — APPROVED** [review]

**Review by MartinEden — APPROVED** [review]
